lib/helpers/model/applyHooks.js

```diff
@@ -164,3 +150,33 @@ function applyHooks(model, schema, options) {
       createWrapper(method, originalMethod, null, customMethodOptions);
   }
 }
+
+/**
+ * Check if there is an embedded schematype in the given schematype. Handles drilling down into primitive
+ * arrays and maps in case of array of array of subdocs or map of subdocs.
+ *
+ * @param {SchemaType} curType
+ * @returns { childModel: Model | typeof Subdocument, curType: SchemaType } | null
+ */
+
+function findChildModel(curType) {
+  if (curType.$isSingleNested) {
+    return { childModel: curType.caster, type: curType };
+  }
+  if (curType.$isMongooseDocumentArray) {
+    return { childModel: curType.Constructor, type: curType };
+  }
+  if (curType.instance === 'Array') {
+    const embedded = curType.getEmbeddedSchemaType();
+    if (embedded) {
+      return findChildModel(embedded);
+    }
+  }
+  if (curType.instance === 'Map') {
```

Comment:
Unless i missed something, this did not exist in the original definition, right?

Reply:
Correct, the original definition did not have support for this. I'll see if we can add this `findChildModel()` function to master.